src/FieldsBuilder.php

```diff
@@ -824,4 +853,220 @@ private function getPrefetchMethodInfo(ReflectionClass $refClass, $reflector, ob
 
         return [$prefetchMethodName, $prefetchArgs, $prefetchRefMethod];
     }
+
+    /**
+     * Gets input fields by class method annotations.
+     *
+     * @param string|object $controller
+     * @param class-string<AbstractRequest> $annotationName
+     * @param array<mixed> $defaultProperties
+     *
+     * @return array<string, InputField>
+     *
+     * @throws AnnotationException
+     */
+    private function getInputFieldsByMethodAnnotations($controller, ReflectionClass $refClass, ReflectionMethod $refMethod, string $annotationName, bool $injectSource, array $defaultProperties, ?string $typeName = null, bool $isUpdate = false): array
+    {
+        $fields = [];
+
+        $annotations = $this->annotationReader->getMethodAnnotations($refMethod, $annotationName);
+        foreach ($annotations as $fieldAnnotations) {
+            $description = null;
+            if ($fieldAnnotations instanceof Field) {
+                $for = $fieldAnnotations->getFor();
+                if ($typeName && $for && !in_array($typeName, $for)) {
+                    continue;
+                }
+                $description = $fieldAnnotations->getDescription();
+
+                $docBlockObj = $this->cachedDocBlockFactory->getDocBlock($refMethod);
+                $methodName = $refMethod->getName();
+                if (strpos($methodName, 'set') !== 0) continue;
+                $name = $fieldAnnotations->getName() ?: $this->namingStrategy->getInputFieldNameFromMethodName($methodName);
+                if (!$description) {
+                    $description = $docBlockObj->getSummary() . "\n" . $docBlockObj->getDescription()->render();
+                }
+
+                $inputFieldDescriptor = new InputFieldDescriptor();
+                $inputFieldDescriptor->setRefMethod($refMethod);
+                $inputFieldDescriptor->setIsUpdate($isUpdate);
+                $inputFieldDescriptor->setName($name);
+                $inputFieldDescriptor->setComment(trim($description));
+
+                $parameters = $refMethod->getParameters();
+                if ($injectSource === true) {
+                    $firstParameter = array_shift($parameters);
+                    // TODO: check that $first_parameter type is correct.
```

Comment:
Is this needed?  If not, let's update the code comment to something that's more helpful for other devs.

Reply:
this is copied from the getter part of the fields builder which has this comment, from what i understand, there should be a sanity check if the injected source is actually the same Type as hinted.

I'd leave it for now.